planner/core/logical_plan_builder.go

```diff
@@ -4872,7 +4872,28 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	var result LogicalPlan = ds
 	dirty := tableHasDirtyContent(b.ctx, tableInfo)
 	if dirty || tableInfo.TempTableType == model.TempTableLocal || tableInfo.TableCacheStatusType == model.TableCacheStatusEnable {
-		us := LogicalUnionScan{handleCols: handleCols}.Init(b.ctx, b.getSelectOffset())
+		var uniqueIndexCols []UniqueIndexCols
+		for _, idx := range ds.table.Indices() {
+			if !tables.IsIndexWritable(idx) {
+				continue
+			}
+			idxInfo := idx.Meta()
+			if !idxInfo.Unique {
+				continue
+			}
+			if ds.table.Meta().IsCommonHandle && idxInfo.Primary {
```

Comment:
Do we need to check `PKIsHandle` here?

Reply:
Sorry I can't get you, how to check `PKIsHandle` here?

Reply:
Like 
```suggestion
			if (ds.table.Meta().IsCommonHandle || ds.table.Meta().PKIsHandle) && idxInfo.Primary {
```

Reply:
It's ok here, `table.Indices()` dosn't include `PKIsHandle`.